sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -181,6 +181,13 @@ public abstract partial class Response<T>
         public static implicit operator T (Azure.Response<T> response) { throw null; }
         public override string ToString() { throw null; }
     }
+    public static partial class SerializationExtensions
+    {
+        public static System.BinaryData SerializeToBinaryData(this Azure.Core.Serialization.ObjectSerializer serializer, object? value, System.Type? inputType = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```

Comment:
I think this is going to be a heavily used API.  What do you think about shortening the name to just `Serialize`?  I'm on the fence because then we're changing the return type based on whether or not the customer passes a `Stream` and that's kind of subtle, but on the other hand I think more folks will find/use BinaryData if they try to call `Serialize` and this is easier than trying to find an output stream to use.

Reply:
Agreed, especially if we move it to the `ObjectSerializer` itself.